### PR TITLE
 Separate IWYU pragma lines from include lines

### PR DESCRIPTION
**Pull request by wthrowe**
### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Review by kidder — DISMISSED** [review]

**Review by nilsdeppe — DISMISSED** [review]
@wthrowe there is one legitimate clang-tidy issue in `IsentropicVortexSource` but that's not related to this PR directly. I'm fine ignoring that for this PR, but wanted to check what you would prefer before we merge :)

**Comment by wthrowe** [comment]
It's a false positive.  I'll throw in a NOLINT.

The IWYU timeout is due to Test_IndexManipulation.cpp taking a ridiculously long time for some reason.

**Review by nilsdeppe — APPROVED** [review]
Thanks @wthrowe!

**Comment by nilsdeppe** [comment]
Travis failures:
- clang-tidy: boost internals
- IWYU: times out :man_shrugging:

File: src/NumericalAlgorithms/DiscontinuousGalerkin/Actions/ApplyBoundaryFluxesLocalTimeStepping.hpp [hunk 1]
```diff
@@ -16,10 +16,11 @@
 #include "NumericalAlgorithms/DiscontinuousGalerkin/Tags.hpp"  // IWYU pragma: keep // for db::item_type<Tags::Mortars<...>>
 #include "Time/Tags.hpp"  // IWYU pragma: keep // for db::item_type<Tags::TimeStep>
 #include "Time/TimeSteppers/TimeStepper.hpp"  // IWYU pragma: keep
-// IWYU pragma: no_include "Time/Time.hpp" // for TimeDelta
 #include "Utilities/Gsl.hpp"
 #include "Utilities/TaggedTuple.hpp"
 
+// IWYU pragma: no_include "Time/Time.hpp" // for TimeDelta
+
```
Comment (mar-celine) [inline src/NumericalAlgorithms/DiscontinuousGalerkin/Actions/ApplyBoundaryFluxesLocalTimeStepping.hpp:23]: On line 25 should the IWYU pragma come before the cond?

**Review by mar-celine — COMMENTED** [review]
I think in the future we should add to the documentation how we want the IWYU pragmas organized, similarly to how we currently outline the include order.

File: src/NumericalAlgorithms/DiscontinuousGalerkin/Actions/ApplyBoundaryFluxesLocalTimeStepping.hpp (hunk 1 above)
Comment (wthrowe) [reply src/NumericalAlgorithms/DiscontinuousGalerkin/Actions/ApplyBoundaryFluxesLocalTimeStepping.hpp:23]: Maybe?  I don't think we have a style consensus for that.  It seems unrelated to this PR in any case.

Comment (nilsdeppe) [reply src/NumericalAlgorithms/DiscontinuousGalerkin/Actions/ApplyBoundaryFluxesLocalTimeStepping.hpp:23]: I agree that there is no consensus (or even a discussion as far as I'm aware) and that deferring that to a separate PR/issue is better.

**Comment by nilsdeppe** [comment]
@gsb76 please respond to the discussion and give your feedback.

File: src/NumericalAlgorithms/DiscontinuousGalerkin/Actions/ApplyBoundaryFluxesLocalTimeStepping.hpp (hunk 1 above)
Comment (mar-celine) [reply src/NumericalAlgorithms/DiscontinuousGalerkin/Actions/ApplyBoundaryFluxesLocalTimeStepping.hpp:23]: I have no preference, and I'm fine leaving the conds where they are so I don't think I'll be opening an issue to change it/ put into writing that there is no consensus.

**Review by mar-celine — APPROVED** [review]

**Comment by kidder** [comment]
ignoring the travis failures of IWYU timeout, and clang-tidy of boost

**Pull request merged**